api/src/main/java/org/openmrs/api/ProgramWorkflowService.java

```diff
@@ -530,4 +535,36 @@ public void purgeConceptStateConversion(ConceptStateConversion conceptStateConve
 	 * @should throw an error when multiple program states with same uuid are found
 	 */
 	public ConceptStateConversion getConceptStateConversionByUuid(String uuid);
+
+        @Transactional(readOnly = true)
+        @Authorized({"Get Program Attribute Types"})
```

Comment:
Since we can have other non patient categories of programs, can we append patient to the privileges? e.g "Get Patient Program Attribute Types" instead of "Get Program Attribute Types"
The same applies to the rest below.

Reply:
okay